### PR TITLE
Feature/init nss wrapper

### PR DESCRIPTION
**Pull request by antevens**
Adds a function to handle all the things needed to use NSS Wrapper.

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -661,6 +661,54 @@ run_dir="${run_dir:-$(dirname "${script_full_path}")}"
...
+function init_nss_wrapper {
+    umask_decorator_mask=${NSS_WRAPPED_FILE_MASK:-0002}
+    umask_decorator "${FUNCNAME[0]}" "${@:-}" && return
+
+    debug 8 'Initializing NSS Wrapper'
+    assert test -n "${BUILD_GUID}"
+
+    export TMP_USER="${1:-bob}"
+    export TMP_GROUP="${2:-builders}"
+    tmp_passwd_file="$(mktemp)" && add_on_exit "rm -f '${tmp_passwd_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_passwd_file}"
```
Comment (hrpatel) [inline shtdlib.sh]: Can we add a suffixes to these paths to make troubleshooting easier? 

e.g. 

```
tmp_passwd_file="$(mktemp)-passwd" && ...
```

File: shtdlib.sh [hunk 2]
```diff
@@ -661,6 +661,54 @@ run_dir="${run_dir:-$(dirname "${script_full_path}")}"
...
+    tmp_group_file="$(mktemp)" && add_on_exit "rm -f '${tmp_group_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_group_file}"
+    tmp_hosts_file="$(mktemp)" && add_on_exit "rm -f '${tmp_hosts_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_hosts_file}"
+
+    if [ -n "${3:-}" ] ; then
+        tmp_home_path="${3}"
+    else
+        tmp_home_path="$(mktemp -d)" && add_on_exit "rm -Rf '${tmp_home_path}'" && chown -R "${BUILD_GUID}" "${tmp_home_path}"
```
Comment (hrpatel) [inline shtdlib.sh]: Same as above

File: shtdlib.sh (hunk 1 above)
Comment (antevens) [reply shtdlib.sh]: That's a good idea, I've switched to using templates and including the filename and pid.

File: shtdlib.sh (hunk 2 above)
Comment (antevens) [reply shtdlib.sh]: See above

File: shtdlib.sh (hunk 1 above)
Comment (hrpatel) [reply shtdlib.sh]: Whats the `XXXXXXXXXX` for? Just so we know its our stuff?

File: shtdlib.sh [hunk 3]
```diff
@@ -661,6 +661,54 @@ run_dir="${run_dir:-$(dirname "${script_full_path}")}"
...
+    export TMP_USER="${1:-bob}"
+    export TMP_GROUP="${2:-builders}"
+    tmp_passwd_file="$(mktemp -t "passwd.${$}.XXXXXXXXXX")" && add_on_exit "rm -f '${tmp_passwd_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_passwd_file}"
+    tmp_group_file="$(mktemp -t "group.${$}.XXXXXXXXXX")" && add_on_exit "rm -f '${tmp_group_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_group_file}"
+    tmp_hosts_file="$(mktemp -t "hosts.${$}.XXXXXXXXXX")" && add_on_exit "rm -f '${tmp_hosts_file}'" && chmod "${NSS_WRAPPED_FILE_PERM:-0664}" "${tmp_hosts_file}"
+
+    if [ -n "${3:-}" ] ; then
+        TMP_HOME_PATH="${3}"
+    else
+        TMP_HOME_PATH="$(mktemp -t -d "home.${TMP_USER}.XXXXXXXXXX")" && add_on_exit "rm -Rf '${TMP_HOME_PATH}'" && chown -R "${BUILD_GUID}" "${TMP_HOME_PATH}"
```
Comment (hrpatel) [inline shtdlib.sh]: Technically the `-d` should be before `-t`

Comment (antevens) [reply shtdlib.sh]: I'll swap those around for OSX compat.

Comment (hrpatel) [reply shtdlib.sh]: Shouldn't it be the case for Linux as well?

**Pull request merged**